### PR TITLE
Prevent Dialog scrollbars from showing up during show transition

### PR DESCRIPTION
**Pull request by domidomi**
Related issue: https://github.com/brainly/style-guide/issues/2312

### CODE REVIEW

File: src/components/dialog/Dialog.jsx [hunk 1]
```diff
@@ -97,6 +100,7 @@ function BaseDialog({
 
     if (!supportsTransitions()) {
       fireTransitionEndCallbacks();
+      setHasFinishedTransition(true);
```
Comment (coderitual) [inline src/components/dialog/Dialog.jsx:103]: Why not to put `setHasFinishedTransition(true);` in `fireTransitionEndCallbacks`?

**Review by coderitual — APPROVED** [review]

File: src/components/dialog/Dialog.jsx [hunk 2]
```diff
@@ -100,6 +108,20 @@ function BaseDialog({
...
   }, [open, fireTransitionEndCallbacks]);
 
+  React.useEffect(() => {
+    if (open) {
+      const dialogHeight =
+        containerRef.current &&
+        containerRef.current.getBoundingClientRect().height;
+
+      const overlayHeight =
+        overlayRef.current && overlayRef.current.getBoundingClientRect().height;
```
Comment (coderitual) [inline src/components/dialog/Dialog.jsx:118]: maybe just `overlayRef?.current.getBoundingClientRect().height`?

Comment (domidomi) [reply src/components/dialog/Dialog.jsx:118]: @coderitual `containerRef?.current?.getBoundingClientRect().height;` would work

Comment (coderitual) [reply src/components/dialog/Dialog.jsx]: oh sry we probably don't need first condition, so just `containerRef.current?.getBoundingClientRect().height;`

File: src/components/dialog/Dialog.jsx [hunk 3]
```diff
@@ -138,7 +156,9 @@ function BaseDialog({
   );
 
   const overlayClass = cx('js-dialog', 'sg-dialog__overlay', {
-    'sg-dialog__overlay--scroll': scroll === 'outside',
+    'sg-dialog__overlay--scroll':
+      (showScrollbarsInitially || hasFinishedTransition) &&
```
Comment (coderitual) [inline src/components/dialog/Dialog.jsx:160]: @domidomi can we rename `showScrollbarsInitially` to something more meaningful? We can name it by reason or by effect but the next. condition is by reason and this one by effect. We shouldn't mix those 2 appraoches.

**Review by clxandstuff — APPROVED** [review]

**Pull request merged**